### PR TITLE
GUI - Map - Optimisation

### PR DESCRIPTION
**Pull request by osvegn**
This pull request implements some optimization on the map displayed.

### CODE REVIEW

File: src/zappy_gui_src/Core/SFML/Map/Map.cpp [hunk 1]
```diff
@@ -48,21 +51,65 @@ void Map::_updateMoveMap(sf::Vector2f &moveMap)
...
+            i = tmp + _mapSize.x - 2;
+            tmp = 0;
+            continue;
+        }
+        if (area.top + area.height < 0)
+            continue;
+        if (area.top > _window->getSize().y && tmp) {
+            i = tmp + _mapSize.x - 2;
+            tmp = 0;
+            continue;
```
Comment (pariciadra) [inline src/zappy_gui_src/Core/SFML/Map/Map.cpp:94]: I'm not sure but what happen if not ?

File: src/zappy_gui_src/Core/SFML/Map/Map.cpp [hunk 2]
```diff
@@ -48,21 +51,65 @@ void Map::_updateMoveMap(sf::Vector2f &moveMap)
 void Map::display()
```
Comment (pariciadra) [inline src/zappy_gui_src/Core/SFML/Map/Map.cpp:51]: I don't understand what this function do, many call to clock with -1 index and we don't know why.
Can you please split in understandable function, will appreciate that

File: src/zappy_gui_src/Core/SFML/Map/Tile.cpp [hunk 3]
```diff
@@ -39,10 +39,13 @@ Tile Tile::operator=(const Tile &tile)
 
 void Tile::setTexture(const std::string &texturePath)
```
Comment (pariciadra) [inline src/zappy_gui_src/Core/SFML/Map/Tile.cpp:40]: Please do not call a function with an already existing name that you use in the same function

File: src/zappy_gui_src/Core/SFML/Map/Tile.hpp [hunk 4]
```diff
@@ -70,6 +72,10 @@ namespace gui {
                 return _position;
             };
 
+            inline const sf::FloatRect getIsoPosition() {
```
Comment (pariciadra) [inline src/zappy_gui_src/Core/SFML/Map/Tile.hpp:75]: Missing documentation

File: src/zappy_gui_src/Core/SFML/Map/Tile.hpp [hunk 5]
```diff
@@ -100,10 +115,13 @@ namespace gui {
                 return _shape.getFillColor();
             };
 
-            /// \brief Set the texture of the tile.
-            /// \param texturePath The path of the texture to be load.
-            void setTexture(const std::string &texturePath);
+            const std::string &getTexturePath() const {
```
Comment (pariciadra) [inline src/zappy_gui_src/Core/SFML/Map/Tile.hpp:118]: add inline + doc

File: src/zappy_gui_src/Core/SFML/SFML.cpp [hunk 6]
```diff
@@ -19,7 +19,7 @@ SFML::SFML() : _run(true)
     _window.get()->setFramerateLimit(60);
     _map.setWindow(_window);
     _map.setEvent(_event);
-    _map.setMapSize(sf::Vector2f(20, 10));
+    _map.setMapSize(sf::Vector2f(1000, 500));
```
Comment (pariciadra) [inline src/zappy_gui_src/Core/SFML/SFML.cpp:22]: Better way is to use static const int because i don't even know what 1000 is supposed to be

**Review by pariciadra — CHANGES_REQUESTED** [review]
Pls checkout

File: src/zappy_gui_src/Core/SFML/Map/Map.cpp (hunk 1 above)
Comment (osvegn) [reply src/zappy_gui_src/Core/SFML/Map/Map.cpp:94]: laaaaaaaaaaaaaaag

Comment (osvegn) [reply src/zappy_gui_src/Core/SFML/Map/Map.cpp:94]: all tile with a position in one of these condition are not allowed to be display, so they are skipped to increase the performance

File: src/zappy_gui_src/Core/SFML/Map/Tile.cpp (hunk 3 above)
Comment (osvegn) [reply src/zappy_gui_src/Core/SFML/Map/Tile.cpp:40]: maybe "setThePathNameOfTheTextureAndSetTheTextureOnTheShapeAndCreateAsharedPointOfTheTexture", but it's too long, so "setTexture" is a good name, doesn't it ?

**Review by LucasTesnier — APPROVED** [review]
NTR!

**Pull request merged**